packer/plugin_folders.go

```diff
@@ -7,33 +7,23 @@ import (
 	"log"
 	"os"
 	"path/filepath"
-	"strings"
 
 	"github.com/hashicorp/packer-plugin-sdk/pathing"
 )
 
-// PluginFolders returns the list of known plugin folders based on system.
-func PluginFolders(dirs ...string) []string {
-	res := []string{}
+// PluginFolder returns the known plugin folder based on system.
+func PluginFolder() (string, error) {
+	var res string
 
 	if packerPluginPath := os.Getenv("PACKER_PLUGIN_PATH"); packerPluginPath != "" {
-		res = append(res, strings.Split(packerPluginPath, string(os.PathListSeparator))...)
-		return res
+		return packerPluginPath, nil
 	}
 
-	if path, err := os.Executable(); err != nil {
-		log.Printf("[ERR] Error finding executable: %v", err)
-	} else {
-		res = append(res, filepath.Dir(path))
-	}
-
-	res = append(res, dirs...)
-
-	if cd, err := pathing.ConfigDir(); err != nil {
+	cd, err := pathing.ConfigDir()
+	if err != nil {
 		log.Printf("[ERR] Error loading config directory: %v", err)
-	} else {
-		res = append(res, filepath.Join(cd, "plugins"))
+		return res, err
```

Comment:
Given that we are returning an error I would expect this to be an empty string literal. It's possible that `res` is empty at this time but if something changes in the future and res is not empty I can see this as a bit of a surprise. 
```suggestion
		return "", err
```

Reply:
That's a good call, `res` is actually unused elsewhere in the code, so we can ditch it and return an explicit empty string here.